app/battles/battle.py

```diff
@@ -0,0 +1,27 @@
+from app.battles.knights_preparing import preparing
+
+
+def battle(knights_config: dict) -> dict:
+    battle_result = {}
+    pairs = [("lancelot", "mordred"), ("arthur", "red_knight")]
+    for first_duel, second_duel in pairs:
+        first_duel_result = preparing(knights_config[first_duel])
+        second_stats_result = preparing(knights_config[second_duel])
+
+        first_duel_result["hp"] -= \
+            (
+                second_stats_result["power"] - first_duel_result["protection"]
+        )
+        if first_duel_result["hp"] <= 0:
+            first_duel_result["hp"] = 0
+        second_stats_result["hp"] -= \
+            (
+                first_duel_result["power"] - second_stats_result["protection"]
+        )
```

Comment:
Do not use line breaks without a need, reformat your code to not use line breaks `\`.